src/engraving/rendering/dev/beamlayout.cpp

```diff
@@ -1247,7 +1258,27 @@ void BeamLayout::createBeamSegment(Beam* item, ChordRest* startCr, ChordRest* en
             }
 
             int extraBeamAdjust = cr->up() ? beamsBelow : beamsAbove;
-            addition = grow * (level - extraBeamAdjust) * item->beamDist();
+            int stemShortenFactor = 0;
+            if (shortStems && cr != item->elements().front() && cr != item->elements().back()) {
+                if (cr == startCr || cr == endCr) {
+                    if (b->above != cr->up()) {
+                        bool change = true;
+                        for (ChordRest* alteredCr : alteredCrs) {
+                            if (cr == alteredCr) {
+                                change = false;
+                                break;
+                            }
+                        }
+                        if (change) {
+                            stemShortenFactor = level - 1;
+                        }
```

Comment:
If I'm not mistaken, I think all of this can be done more simply:
```
if (!mu::contains(alteredCrs, cr) {
    stemShortenFactor = level - 1;
}
```

Reply:
Oh! That's convenient, thank you